lib/volk_rank_archs.c

```diff
@@ -34,13 +34,16 @@ int volk_get_index(const char* impl_names[], // list of implementations by name
 {
     unsigned int i;
     for (i = 0; i < n_impls; i++) {
-        if (!strncmp(impl_names[i], impl_name, 20)) {
+        if (!strncmp(impl_names[i], impl_name, 42)) {
             return i;
         }
     }
     // TODO return -1;
     // something terrible should happen here
     fprintf(stderr, "Volk warning: no arch found, returning generic impl\n");
+    if (strncmp(impl_name, "generic", 20)) {
```

Comment:
can we use 7 or 8 instead of 20? "generic" is 7 bytes long, so 20 way more than necessary ...

Reply:
IDK whether we use 7 or 8 bytes ... would have to test `strncmp` to see ...

Reply:
Also: If we're adding in this check, then let's save the result of this `strncmp` and return it instead of the recursive call to this function. Once through and done.

Reply:
If we change the recursive structure, we'd need to change the whole function quite significantly.
Right now, we follow a 3 step process:
1. Search for the user requested impl
2. If user request cannot be fulfilled: Search for generic impl.
3. New third step: Give up and return `-1`.
I'm actually not entirely sure, if the term `generic` is the only thing we ever search for. It might be smth like `vector_generic` as well. However, not during the second pass.

Right now, it feels like we actually need 2 functions. One with the loop and one to handle all the different error conditions.